### ISSUE TITLE
loss become nan, invalid value encountered in log

### ISSUE THREAD

**Issue by jycloud** [eragonruan/text-detection-ctpn#49]
text-detection-ctpn-master1/lib/fast_rcnn/bbox_transform.py:29: RuntimeWarning: invalid value encountered in log
  targets_dh = np.log(gt_heights / ex_heights)

in my training process,this warning come out and the loss become nan. why i get this error?

**Comment by eragonruan** [comment]
@jycloud `targets_dh = np.log(gt_heights / ex_heights)` bug occurs when gt_heights=0, see [py-faster-rcnn/issues/19](https://github.com/rbgirshick/py-faster-rcnn/issues/19), this may caused by the annotation of your training data

**Comment by jycloud** [comment]
@eragonruan I check my training data by cv2.retangle() to show the coordinate,it's all right. I used ICDAR dataset to train ctpn,this error came out as same,I insert the code (if gt_heights<=0,print gt_heights,)to check, the program print [-32. -32. -32. ..., -32. -32. -32.].

**Comment by eragonruan** [comment]
@jycloud can you train the model for several step, or the loss become nan at the first step. if you can train for several step, set DISPLAY to 1 in text.yml and print out the filename when training to locate the bug, then check the annotation file. Also, did you convert the annotation to x1,y1,x2,y2,x3,y3,x4,y4.

**Comment by jycloud** [comment]
@eragonruan I rerun split_label.py code ,it solve my problem, it's weird.Thank you for your answer.And I have another question, I find the results is not good enough in small words case,do you have some suggestion to improve the performance?

**Comment by eragonruan** [comment]
@jycloud you can start a new issue to discuss this.
